gapic/samplegen_utils/utils.py

```diff
@@ -81,7 +87,7 @@ def is_valid_sampleconfig_p(doc) -> bool:
 
         for fullpath in yaml_file_generator:
             with open(fullpath) as f:
-                if all(is_valid_sampleconfig_p(doc)
+                if any(is_valid_sample_cfg(doc)
```

Comment:
Where do you check that across all the files thus yielded, at least one of them is a sample spec? It might be helpful to mention that here in a comment.

Reply:
Any file yielded has at least one yaml doc that is a sample spec. The `yaml.safe_load_all` iterates over all the yaml docs in the file.

Reply:
The requirement is that at the end of the day, after we process all files and directories, we end up with one or more  sample specs. This means that: 

1. if only directories are specified (not files) then there should be at least one sample spec doc being yielded by the set of all files in all the specified directories. If there isn't, we error.
2. If files and directories are both specified, it's OK if no sample docs come from any of the directories, because we know the specified files must provide some as per the checks above.

I see 2 works, but where do you check 1? (Sorry if I missed it)

Reply:
Addressed in upcoming commit.